lib/iris/fileformats/netcdf.py

```diff
@@ -759,7 +759,49 @@ def coord_from_term(term):
         cube.add_aux_factory(factory)
 
 
-def load_cubes(filenames, callback=None):
+def _translate_constraints_to_var_callback(constraints):
+    """
+    Translate load constraints into a simple data-var filter function, if possible.
+
+    Returns:
+         * function(cf_var:CFDataVariable): --> bool,
+            or None.
+
+    For now, ONLY handles a single NameConstraint with no 'STASH' component.
+
+    """
+    import iris._constraints
+
+    constraints = iris._constraints.list_of_constraints(constraints)
+    result = None
+    if len(constraints) == 1:
+        (constraint,) = constraints
+        if (
+            isinstance(constraint, iris._constraints.NameConstraint)
+            and constraint.STASH == "none"
+        ):
+            # As long as it doesn't use a STASH match, then we can treat it as
+            # a testing against name properties of cf_var.
+            # That's just like testing against name properties of a cube, except that they may not all exist.
+            def inner(cf_datavar):
+                match = True
+                for name in constraint._names:
+                    expected = getattr(constraint, name)
+                    if name != "STASH" and expected != "none":
+                        attr_name = "cf_name" if name == "var_name" else name
+                        # Fetch property : N.B. CFVariable caches the property values
+                        # The use of a default here is the only difference from the code in NameConstraint.
+                        actual = getattr(cf_datavar, attr_name, "")
```

Comment:
@lbdreyer Is there an edge case here, where we've set the default `actual` value to be `""` for an `attr_name` that doesn't exist on the `cf_datavar`, but that could match an `expected` of `""`, leaving `match = True` but instead it should actually be `False`.

Is it safer to skip the `actual` and `expected` value comparison if the `cf_datavar` doen't have an `attr_name`? i.e.,

```python
if not hasattr(cf_datavar, attr_name):
    continue
actual = getattr(cf_datavar, attr_name)
...
```
Not sure whether this is relevant here... just a thought :thinking: